src/checkpoint/buffer/io_buffer.cc

```diff
@@ -59,8 +59,73 @@
 
 namespace checkpoint { namespace buffer {
 
-void IOBuffer::setupFile() {
-  debug_checkpoint("IOBuffer: opening file: %s\n", file_.c_str());
+void IOBuffer::setupForRead() {
+  debug_checkpoint("IOBuffer: opening file for read: %s\n", file_.c_str());
+
+  int ret = 0;
+
+  /*
+   * Start by opening the file, read-only mode
+   */
+  fd_ = open(file_.c_str(), O_RDONLY, (mode_t)0600);
+  if (fd_ == -1) {
+    auto err = std::string("Failed to open file=") + file_ + ": errno=" +
+      std::to_string(errno);
```

Comment:
You could include `perror(errno)` in these strings, so that one doesn't need to look up meanings.

Reply:
Ok, I've added `strerror(errno` to the runtime error (can't use perror because it just prints to stderr, doesn't append to my error mesage).